File: Data Structures & Algorithms/minimum-window-with-characters/submission-0.cpp
class Solution {
public:
    string minWindow(string s, string t) {
        unordered_map<char, int> table;
        int need = t.size();
        for (int i = 0; i < need; i++) table[t[i]]++;
        int s_size = s.size();
        int left = 0, start = 0;
        int min_len = INT_MAX;
        for (int right = 0; right < s_size; right++)
        {
            if (table[s[right]] > 0)
                need--;
            table[s[right]]--;
            while (need == 0)
            {
                if (right - left + 1 < min_len)
                {
                    min_len = right - left + 1;
                    start = left;
                }
                table[s[left]]++;
                if (table[s[left]] > 0)
                    need++;
                left++;
            }
        }
        return min_len == INT_MAX ? "" : s.substr(start, min_len);
    }
};

File: Data Structures & Algorithms/daily-temperatures/submission-0.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        mp = [0] * len(temperatures)
        for left in range (len(temperatures)):
            # right = left + 1
            for right in range(left, len(temperatures)):
                if temperatures[left] < temperatures[right]:
                    mp[left] = right - left
                    break
            if right >= len(temperatures):
                mp[left] = 0
        return mp